framework/src/utils/PetscSupport.C

```diff
@@ -136,16 +137,26 @@ setSolverOptions(SolverParams & solver_params)
   }
 }
 
-void petscSetupDM (NonlinearSystem & nl) {
+int petscSetupDM (NonlinearSystem & nl) {
```

Comment:
Why did you change this function to return an `int`?  It never returns anything other than zero...